src/error.rs

```diff
@@ -0,0 +1,82 @@
+use crate::{snapshot, strval::StrValError};
+
+use soroban_env_host::{
+    xdr::{Error as XdrError, ScVal},
+    HostError,
+};
+use soroban_spec::gen::{json, rust};
+
+use hex::FromHexError;
+use std::io;
+
+#[derive(thiserror::Error, Debug)]
+pub enum CmdError {
```

Comment:
Moving these errors into a single place makes it difficult to ascertain what error conditions exist for each sub-command. The errors were listed in each subcommand as a goal in improving the ability to understand the error conditions of each command we add. What's the motivation for moving all the errors into a single file?

Reply:
Multiple commands use the same errors, contract id parsing, ledger file opening etc.

Reply:
There are some duplicates, but I don't think we need to have complete code deduplication. As this tool grows this error list will grow, and I think it will become difficult to manage a large list of unrelated errors.

Reply:
I think we can improve the error reporting without deduplicating the errors into a single list?

Reply:
I suppose I can defactor it.

Reply:
PTAL